File: CombatTracker-Application/CombatTrackerData/Actions/DeathAction.vb
Partial Class DeathAction
    Public Sub New(ByVal TheChar As Actor, ByVal CurrentTime As Double)
        WhoIsActing = TheChar
        Me.Name = "Unconcious"
        Me.Note = ""
        Me.BasePercent = 1
        Me.StartTime = CurrentTime
        Me.EndTime = TheChar.CalculateTimeRequiredNonEncumbered(BaseRoundTime) + CurrentTime
    End Sub
    Public Overrides ReadOnly Property Reoccuring() As Boolean
        Get
            Return WhoIsActing.HitsRemaining <= 0
        End Get
    End Property
    Public Overrides ReadOnly Property CharacterAction() As Boolean
        Get
            Return True
        End Get
    End Property

    Public Overrides Sub HandleAction(ByVal DB As RMSSDataDataContext, ByVal GI As Game)
        If WhoIsActing.HitsRemaining < -WhoIsActing.ConstitutionStat Then
            Me.Name = WhoIsActing.Name & " is Dead"
            Count += 1
            Note = "Rounds of Death:" & Count
        End If
        Me.EndTime = WhoIsActing.CalculateTimeRequiredNonEncumbered(BaseRoundTime) + GI.CurrentTime
        'WhoIsActing.HandleInitChange(MD)
    End Sub

    Public Overrides Sub RefreshData()

    End Sub

    Private Sub OnCreated()

        Me.BasePercent = 0.001
        Name = ""
        StartTime = 0
        EndTime = 0
        Note = ""
        State = ActionProblem.IsOK
        CurrentModifier = 0
        Type = ActorActionType.Death
    End Sub
End Class

File: Previous Rewrite/CombatTracker.DomainOld/Current/Actions/DeathAction.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using CombatTracker.Domain.Reference;

namespace CombatTracker.Domain.Current.Actions
{
    public class DeathAction : BaseAction
    {
        public int Count { get; set; }


        public DeathAction(Actor theChar, double currentTime)
            : base(currentTime)
        {
            WhoIsActing = theChar;
            Name = "Unconcious";
            StartTime = currentTime;
            BasePercent = 1;
            Type = ActorActionType.Death;
            EndTime = WhoIsActing.CalculateTimeRequiredNonEncumbered(MySettings.BaseRoundTime) + currentTime;
        }

        public override bool CharacterAction
        {
            get { return true; }
        }

        public override void HandleAction(RmssDbContext db, double currentTime)
        {
            if (WhoIsActing.HitsRemaining < -WhoIsActing.ConstitutionStat)
            {
                Name = WhoIsActing.Name + " is Dead";
                Count += 1;
                Note = "Rounds of Death:" + Count;
            }
            EndTime = WhoIsActing.CalculateTimeRequiredNonEncumbered(MySettings.BaseRoundTime) + currentTime;
        }

        public override void RefreshData()
        {
            
        }
        public override bool Reoccuring
        {
            get { return WhoIsActing.HitsRemaining <= 0; }
        }
    }
}
